docs/concepts/pools/managed.md

```diff
@@ -71,6 +71,71 @@ uint256[] memory tokenWeights = _managedPool.getNormalizedWeights();
+
+### Removing Tokens
+`removeToken` removes a token from the Pool's list of swappable assets. When removing a token from a Managed Pool, the weights of all other tokens in the pool will increase. Because pools must have at least two tokens (excluding their own BPT), `owner`s can remove a token from the pool as long as there are three or more tokens currently in the pool. Similar to the rules for adding tokens, token removals are also forbidden during weight changes or when a weight change is scheduled in the future. Before calling `removeToken`, the `assetManager` must have withdrawn the entire balance of the token being removed from the pool.
+
+### Examples
+[ManagedPoolAddRemoveTokenLib.sol](https://github.com/balancer/balancer-v2-monorepo/blob/master/pkg/pool-weighted/contracts/managed/ManagedPoolAddRemoveTokenLib.sol) provides the necessary logic for adding and removing tokens from a Managed Pool. [AssetManagers.sol](https://github.com/balancer/balancer-v2-monorepo/blob/master/pkg/vault/contracts/AssetManagers.sol) provides the necessary logic for depositing and withdrawing tokens from a Managed Pool. Below are a few basic examples of how an `owner` can add and remove tokens from a Managed Pool, as well as how an `assetManager` can deposit and withdraw tokens.
+
+```solidity
+// Variable declarations
+IERC20 token = IERC20(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2); // WETH
+address assetManager = 0x123456789....; 
+uint256 tokensNormalizedWeight = 10e16; // 10% normalized weight
+uint256 mintAmount = 100e18;
+uint256 depositAmount = 100e18;
+uint256 burnAmount = 100e18;
+```
+```solidity
+/* Add token to the pool */
+_managedPool.addToken(
+  token,
+  assetManager,
+  tokensNormalizedWeight,
+  mintAmount,
+  msg.sender
+);
+
+// Approve the vault to spend the token
+token.approve(address(vault), depositAmount);
+
+// Deposit token via the assetManager
+IVault.PoolBalanceOp[] memory ops = new IVault.PoolBalanceOp[](1);
+ops[0] = IVault.PoolBalanceOp({
+  kind: IVault.PoolBalanceOpKind.Deposit,
+  token: token,
+  amount: depositAmount
+});
+vault.managePoolBalance(ops);
+```
+```solidity
+/* Remove token from the pool */
+// Get current balance of token in the pool
+(uint256 balance, , , ) = vault.getPoolTokenInfo(poolId, token);
+
+// Withdraw token via the assetManager
+IVault.PoolBalanceOp[] memory ops = new IVault.PoolBalanceOp[](1);
+ops[0] = IVault.PoolBalanceOp({
+  kind: IVault.PoolBalanceOpKind.Withdraw,
+  token: token,
+  amount: balance
+});
+vault.managePoolBalance(ops);
```

Comment:
I actually think this won't work (yet). `Withdraw` will move all of the `cash` balance to `managed` balance so the pool will still think that it has a balance. You still need to do an `Update` operation to tell the Vault that you've taken everything out. I think you need to do something like the following:

```
(uint256 cash, uint256 managed, , ) = vault.getPoolTokenInfo(poolId, token);
uint256 total = cash + managed;
IVault.PoolBalanceOp[] memory ops = new IVault.PoolBalanceOp[](2);
ops[0] = IVault.PoolBalanceOp({
  kind: IVault.PoolBalanceOpKind.Withdraw,
  token: token,
  amount: balance
});
ops[1] = IVault.PoolBalanceOp({
  kind: IVault.PoolBalanceOpKind.Update,
  token: token,
  amount: total
});
vault.managePoolBalance(ops);
```